.travis.yml

```diff
@@ -0,0 +1,93 @@
+    unzip $SONARSCANNER.zip
+    rm -f $SONARSCANNER.zip
+    make test -json > report.json
+    make coverage 
+    git clone --branch=devops https://github.com/vmware/singleton.git devops
+    str=$(printf '=%.0s' {1..50})
+    echo $str Unit Test Start $str
+    cd $TRAVIS_BUILD_DIR/devops/sonar/ && python ./config_sonar_project.py -ProjectName=$SONAR_ORG-singleton-go-service-$TRAVIS_BRANCH -ProjectKey=$SONAR_ORG-singleton-go-service-$TRAVIS_BRANCH -OrgKey=$SONAR_ORG -QualityGateName=go-service-gate -QualityGateConditions="./quality_gates/go-service.json" -SonarToken=$SONAR_TOKEN
+    cd $TRAVIS_BUILD_DIR && ./$SONARSCANNERINNER/bin/sonar-scanner -Dsonar.sources=. -Dsonar.exclusions=**/testdata/**/*,**/$SONARSCANNERINNER/**/*,**/bindata/bindata.go,tests/**,api/v1/**,api/v2/swagger/**,assets/**,cmd/**,bindata/**,coverage.html -Dsonar.coverage.exclusions=**/testdata/**/*,**/$SONARSCANNERINNER/**/*,**/bindata/bindata.go,tests/**,api/v1/**,api/v2/swagger/**,assets/**,cmd/**,bindata/**,coverage.html -Dsonar.projectName=$SONAR_ORG-singleton-go-service-$TRAVIS_BRANCH -Dsonar.branch.name=master -Dsonar.projectKey=$SONAR_ORG-singleton-go-service-$TRAVIS_BRANCH -Dsonar.host.url=https://sonarcloud.io -Dsonar.organization=$SONAR_ORG -Dsonar.login=$SONAR_TOKEN -Dsonar.go.tests.reportPaths=./report.json -Dsonar.go.coverage.reportPaths=./cover.out
+    echo $str Unit Test End $str
+    git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '\.travis\.yml$' || {
+      echo "No need to check sonar quality gate for only travis configuration updates"
+      exit 
+    }
+    cd $TRAVIS_BUILD_DIR/devops/sonar && sleep 5 && python3 ./check_sonar_qualitygate.py -ProjectKeyPrefixArray=$SONAR_ORG-singleton-go-service-$TRAVIS_BRANCH -HostName=https://sonarcloud.io
+  fi &&
+
+  if [ x${TEST} = xsmoke_test ]; then
+    export PATH=${PATH/:\/usr\/local\/lib\/jvm\/openjdk11\/bin:/:}
+    sudo apt-get update
+    sudo apt-get install -y openjdk-8-jdk
+    export JAVA_HOME="/usr/lib/jvm/java-8-openjdk-amd64"
+    java -version
+    git clone --branch=master https://github.com/vmware/singleton.git server
+    git clone --branch=devops https://github.com/vmware/singleton.git devops
+    cd $TRAVIS_BUILD_DIR/server/g11n-ws && ./gradlew build -x test
+    cp $TRAVIS_BUILD_DIR/devops/deploy/i18n-service/Dockerfile $TRAVIS_BUILD_DIR/server/publish/
+    cd $TRAVIS_BUILD_DIR/server/publish && mv singleton-[0~9]*.jar i18n-service.jar
+    docker build -t singleton .
+    docker run -d -p 8090:8090 -p 8091:8091 --name singleton singleton
+    docker ps
+    sleep 15s
+    mkdir -p $TRAVIS_BUILD_DIR/l10n/bundles
+    cp -r $TRAVIS_BUILD_DIR/devops/autotest/client/go/l10n/bundles $TRAVIS_BUILD_DIR/l10n/
+    docker cp $TRAVIS_BUILD_DIR/l10n singleton:/
+    curl -X GET 'http://localhost:8091/i18n/api/v2/formatting/patterns?language=en&region=TW&scope=dates'        
+    cd $TRAVIS_BUILD_DIR/devops/autotest/client/go/goclienttest
```

Comment:
Why do you run go client tests against go service?

Reply:
Updated it.